### PR TITLE
refactor(manager): Rewritten the manager code to use the new 3.0 cli

### PR DESCRIPTION
**Pull request by ShlomiBalalis**
## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Comment by fruch** [comment]
pylint isn't happy
```
15:05:37  sdcm/mgmt/operator.py:231:4: W0237: Parameter 'cron' has been renamed to 'start_date' in overridden 'OperatorManagerCluster.create_backup_task' method (arguments-renamed)
```

**Comment by fruch** [comment]
@ShlomiBalalis aren't those changes needed also on manager dtest ?

**Comment by ShlomiBalalis** [comment]
> pylint isn't happy
> 
> ```
> 15:05:37  sdcm/mgmt/operator.py:231:4: W0237: Parameter 'cron' has been renamed to 'start_date' in overridden 'OperatorManagerCluster.create_backup_task' method (arguments-renamed)
> ```

Right, I'll add the changes to the operator class as well



> @ShlomiBalalis aren't those changes needed also on manager dtest ?

Yes, another pr will be added soon

**Comment by ShlomiBalalis** [comment]
Tested on longevity as well both on 2.6 and 3.0, passed with flying colors
@fruch I don't think those new precommit failures are related to me, right? I see stuff about coredumps, etc.

**Comment by fruch** [comment]
> Tested on longevity as well both on 2.6 and 3.0, passed with flying colors @fruch I don't think those new precommit failures are related to me, right? I see stuff about coredumps, etc.

seem like the errors are related to your code:
```
sdcm/mgmt/operator.py:231:4: W0221: Number of parameters was 15 in 'ManagerCluster.create_backup_task' and is now 14 in overridden 'OperatorManagerCluster.create_backup_task' method (arguments-differ)
```

you changed the signature of `create_backup_task` but you need to adapt it also in `OperatorManagerCluster.create_backup_task`, since it's inheriting it.
if you have some version logic, in that function you might need the same for operator (otherwise you might break operator which might be using older version of manager)

**Comment by fruch** [comment]
@vponomaryov FYI, I don't know when those new manager 3.0 change will be coming to the operator. can we add new parameters which are not supported, or operator would break cause of that ?

**Comment by vponomaryov** [comment]
> @vponomaryov FYI, I don't know when those new manager 3.0 change will be coming to the operator. can we add new parameters which are not supported, or operator would break cause of that ?

We can switch to new manager version when it's docker image is available.
But! We are going to have operator functional tests which will cover bunch of manager versions to make sure both work ok in conjuction.
So, I would say that versioning is good approach.
Also, I think that would be great to use similar logic for manager like we have for scylla versions -> https://github.com/scylladb/scylla-cluster-tests/blob/eb6e05ae007a97045e4d8abda948bb97aafdfd0f/sdcm/utils/version_utils.py#L365-L392

File: sdcm/mgmt/cli.py [hunk 1]
```diff
@@ -202,7 +246,12 @@ def progress(self) -> str:
     def detailed_progress(self):
         if self.status in [TaskStatus.NEW, TaskStatus.STARTING]:
             return " 0%"
-        cmd = "task progress {} -c {}".format(self.id, self.cluster_id)
+
+        if self.sctool.parsed_client_version >= new_command_structure_minimum_version:
```
Comment (rayakurl) [inline sdcm/mgmt/cli.py:250]: seems like a code duplication. Please add a function that checks the version and returns the command and call it here.

File: sdcm/mgmt/cli.py [hunk 2]
```diff
@@ -231,7 +280,12 @@ def detailed_progress(self):
     def duration(self) -> datetime.timedelta:
         if self.status in [TaskStatus.NEW, TaskStatus.STARTING]:
             return duration_to_timedelta(duration_string="0")
-        cmd = "task progress {} -c {}".format(self.id, self.cluster_id)
+
+        if self.sctool.parsed_client_version >= new_command_structure_minimum_version:
```
Comment (rayakurl) [inline sdcm/mgmt/cli.py:284]: same as my comment in line 250

File: sdcm/mgmt/cli.py [hunk 3]
```diff
@@ -327,7 +381,10 @@ def __init__(self, task_id, cluster_id, manager_node):
         ManagerTask.__init__(self, task_id=task_id, cluster_id=cluster_id, manager_node=manager_node)
 
     def get_snapshot_tag(self, snapshot_index=0):
-        command = f" -c {self.cluster_id} task progress {self.id}"
+        if self.sctool.parsed_client_version >= new_command_structure_minimum_version:
```
Comment (rayakurl) [inline sdcm/mgmt/cli.py:384]: same as my comment in line 250

File: sdcm/mgmt/cli.py [hunk 4]
```diff
@@ -339,7 +396,11 @@ def get_snapshot_tag(self, snapshot_index=0):
         return snapshot_tag
 
     def is_task_in_uploading_stage(self):
-        full_progress_string = self.sctool.run("task progress {} -c {}".format(self.id, self.cluster_id),
+        if self.sctool.parsed_client_version >= new_command_structure_minimum_version:
```
Comment (rayakurl) [inline sdcm/mgmt/cli.py:399]: same as my comment in line 250

File: sdcm/mgmt/cli.py [hunk 5]
```diff
@@ -175,7 +214,12 @@ def progress(self) -> str:
         """
         if self.status in [TaskStatus.NEW, TaskStatus.STARTING]:
             return " 0%"
-        cmd = "task progress {} -c {}".format(self.id, self.cluster_id)
+
+        if self.sctool.parsed_client_version >= new_command_structure_minimum_version:
```
Comment (rayakurl) [inline sdcm/mgmt/cli.py:218]: same as my comment in line 240

File: sdcm/mgmt/cli.py [hunk 6]
```diff
@@ -213,20 +213,24 @@ def arguments(self) -> str:
                 break
         return arguments_string
 
+    def progress_string(self, **kwargs):
```
Comment (rayakurl) [inline sdcm/mgmt/cli.py:216]: please add a docstring explaining what this funcion is doing and what will be returned.

File: sdcm/mgmt/cli.py [hunk 7]
```diff
@@ -213,20 +213,24 @@ def arguments(self) -> str:
...
+        if self.sctool.is_v3_cli:
+            cmd = f" -c {self.cluster_id} progress {self.id}"
+        else:
+            cmd = f" -c {self.cluster_id} task progress {self.id}"
+
```
Comment (rayakurl) [inline sdcm/mgmt/cli.py:221]: remove the empty line

File: sdcm/mgmt/cli.py [hunk 8]
```diff
@@ -213,20 +213,24 @@ def arguments(self) -> str:
...
+        res = self.sctool.run(cmd=cmd, **kwargs)
+
```
Comment (rayakurl) [inline sdcm/mgmt/cli.py:223]: remove the empty line

**Review by fgelcer — DISMISSED** [review]
LGTM

File: sdcm/mgmt/cli.py [hunk 9]
```diff
@@ -108,8 +122,13 @@ def next_run(self):
         # │ healthcheck/7fb6f1a7-aafc-4950-90eb-dc64729e8ecb │ 18 Nov 18 20:32:08 UTC (+15s) │ 0    │            │ NEW    │
         # │ repair/22b68423-4332-443d-b8b4-713005ea6049      │ 19 Nov 18 00:00:00 UTC (+7d)  │ 3    │            │ NEW    │
         # ╰──────────────────────────────────────────────────┴───────────────────────────────┴──────┴────────────┴────────╯
-        cmd = "task list -c {}".format(self.cluster_id)
+        if self.sctool.is_v3_cli:
+            cmd = "tasks -c {}".format(self.cluster_id)
+        else:
+            cmd = "task list -c {}".format(self.cluster_id)
```
Comment (rayakurl) [inline sdcm/mgmt/cli.py:128]: there is a code duplication between here and lines 162-165

File: sdcm/mgmt/cli.py [hunk 10]
```diff
@@ -140,7 +159,10 @@ def status(self) -> str:
         """
         Gets the task's status
         """
-        cmd = "task list -c {}".format(self.cluster_id)
+        if self.sctool.is_v3_cli:
+            cmd = "tasks -c {}".format(self.cluster_id)
+        else:
+            cmd = "task list -c {}".format(self.cluster_id)
```
Comment (rayakurl) [inline sdcm/mgmt/cli.py:165]: there is a code duplication between here and lines 125-128

**Review by fruch — APPROVED** [review]
LGTM

**Pull request merged**